File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        current = head
        seen = set()
        while current:
            if current in seen:
                return True
            seen.add(current)
            current=current.next
        return False
            
            
        

File: 141-linked-list-cycle/141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode current = head;
        Set<ListNode> seen = new LinkedHashSet<>();
        while(current != null) {
            if(seen.contains(current))  return true;
            seen.add(current);
            current=current.next;
        }
        return false;
        
    }
}